packages/edit-post/src/components/sidebar/post-template/hook.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { store as coreStore } from '@wordpress/core-data';
+import { useMemo, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../../store';
+
+export default function usePostTemplateForm() {
+	const isViewable = useSelect( ( select ) => {
+		const { getCurrentPostType } = select( editorStore );
+		const { getPostType } = select( coreStore );
+		return getPostType( getCurrentPostType() )?.viewable ?? false;
+	}, [] );
+
+	const settings = useSelect(
+		( select ) => select( editorStore ).getEditorSettings(),
+		[]
+	);
+
+	const canUserCreateTemplate = useSelect(
+		( select ) => select( coreStore ).canUser( 'create', 'templates' ),
+		[]
+	);
```

Comment:
Good tip about `useResourcePermissions`. I think it's overkill here while we're only checking `create`.

I didn't want to change any of the `PostTemplatePanel` logic as part of this PR. We only check `create` in `trunk` so that's what I'm doing here. I'm not sure if we should check `edit`. Probably a good idea to investigate that separately.